src/GameLibs/Snake/System/HeadCollision.cpp

```diff
@@ -0,0 +1,53 @@
+/*
+** EPITECH PROJECT, 2023
+** Arcade
+** File description:
+** SFML_EventHandler
+*/
+
+#include "HeadCollision.hpp"
+#include "Snake.hpp"
+#include "SnakeCompType.hpp"
+#include "Sprite.hpp"
+
+bool checkCollision(Arcade::ECS::IComponent &fst, Arcade::Graph::Sprite &head)
+{
+    Arcade::Graph::Sprite &fstS = static_cast<Arcade::Graph::Sprite&>(fst);
+
+    if((head.pos.x >= fstS.pos.x + fstS.rect.width)
+        || (head.pos.x + head.rect.width <= fstS.pos.x)
+        || (head.pos.y >= fstS.pos.y + fstS.rect.height)
+        || (head.pos.y + head.rect.height <= fstS.pos.y)) {
+        return false;
+    } else {
+        return true;
+    }
+}
+
+void Snake::System::HeadCollision::run(float deltaTime, Arcade::ECS::IEventManager &eventManager,
+    Arcade::ECS::IEntityManager &currentScene)
+{
+    std::shared_ptr<Arcade::ECS::IEntity> head = currentScene.getEntitiesById("head");
+    std::unique_ptr<std::vector<std::shared_ptr<Arcade::ECS::IEntity>>> bodies = currentScene.getEntitiesByComponentType(Arcade::ECS::CompType::MOVEABLE);
+    Arcade::ECS::IComponent &apple = currentScene.getEntitiesById("apple")->getComponents("sprite");
+    Arcade::Graph::Sprite &headS = static_cast<Arcade::Graph::Sprite&>(*(head->getComponents(Arcade::ECS::CompType::SPRITE).front().get()));
```

Comment:
static_cast d'une reference d'un dereferencement ? sympa

tu peux direct le cast and shared_ptr non ?